### PR TITLE
Murisi/tx result extraction

### PR DESCRIPTION
**Pull request by murisi**
An attempt to resolve issue #634 . An implementation of a command line option `tx-result` to extract the results of a transaction from its hash. More precisely a transaction result comprises its `info`, `height`, `hash`, `code`, `gas_used`, and `initialized_accounts`. Note also that the hash can either be an application or an acceptance hash.

The following changes were made:
* Add CLI option `tx-result` to allow user to request result
* Made function that uses ABCI/++ query mechanisms to extract results
* A correction to `find_tx` to prevent it from crashing on ABCI++

Note that currently for this patch to work with ABCI++, heliaxdev/tendermint-rs#2 needs to be applied to `tendermint-rs`.

### CODE REVIEW

File: apps/src/lib/cli.rs [hunk 1]
```diff
@@ -673,6 +677,25 @@ pub mod cmds {
...
+
+        fn parse(matches: &ArgMatches) -> Option<Self> {
+            matches
+                .subcommand_matches(Self::CMD)
+                .map(|matches| TxResult(args::TxResult::parse(matches)))
+        }
+
+        fn def() -> App {
+            App::new(Self::CMD)
+                .about("Get the result associated with a transaction.")
```
Comment (tzemanovic) [inline apps/src/lib/cli.rs:694]:
```suggestion
                .about("Get the result of a transaction.")
```

File: apps/src/lib/cli.rs [hunk 2]
```diff
@@ -142,6 +142,7 @@ pub mod cmds {
                 .subcommand(TxUpdateVp::def().display_order(1))
                 .subcommand(TxInitAccount::def().display_order(1))
                 .subcommand(TxInitValidator::def().display_order(1))
+                .subcommand(TxResult::def().display_order(1))
```
Comment (tzemanovic) [inline apps/src/lib/cli.rs:145]: I think this might be a bit better fix for the queries group 3 below

**Review by tzemanovic — APPROVED** [review]
This is looking great, thanks! I think we'll need to print the transactions results a bit nicer to make it clearer the inner tx's hash is the one with the result that can be queried, but that's for another issue. Once we merge https://github.com/heliaxdev/tendermint-rs/pull/2, we should update the lock file to the new ref here

File: apps/src/lib/cli.rs (hunk 2 above)
Comment (murisi) [reply apps/src/lib/cli.rs:145]: Thanks! Good point. I'll rename `TxResult` to `QueryResult`, and place it in `display_order` 3.

**Pull request merged**